dev-vcs/git/git-2.45.2-r1.ebuild

```diff
@@ -630,6 +624,15 @@ src_install() {
 		fi
 	}
 	plocale_for_each_disabled_locale rm_loc
+
+	if ! use safe-directory ; then
+		# Globally setting GIT_IGNORE_INSECURE_OWNER=true neuters
+		# the "safe directory" detection.
+		# bugs #838271, #838223
+		newenvd - 50git-ignore-insecure-owner <<-EOF
+		GIT_IGNORE_INSECURE_OWNER=true
```

Comment:
There is an obvious downside to this compare to the currently approach: It only takes effect once the environment is reloaded.